lib/aws-manager.js

```diff
@@ -81,142 +82,164 @@ class AwsManager {
     let pricingStartDate = new Date();
     pricingStartDate.setMinutes(pricingStartDate.getMinutes() - 30);
 
-    let livingInstances;
-    let livingSpotRequests;
-    let deadInstances;
-    let deadSpotRequests = {};
-    let allPricingHistory;
-    let availableAZ;
-
-    try {
-      livingInstances = await this.ec2.describeInstances({
-        Filters: [
-          {
-            Name: 'key-name',
-            Values: [this.keyPrefix + '*'],
-          },
-          {
-            Name: 'instance-state-name',
-            Values: ['running', 'pending'],
-          },
-        ],
-      });
-      debug('fetched living instance state');
+    this.__previousApiState = this.__apiState;
+    let apiState = {
+      instances: [],
+      requests: [],
+    };
+    let deadState = {
+      instances: [],
+      requests: [],
+    };
 
-      livingSpotRequests = await this.ec2.describeSpotInstanceRequests({
-        Filters: [
-          {
-            Name: 'launch.key-name',
-            Values: [this.keyPrefix + '*'],
-          }, {
-            Name: 'state',
-            Values: ['open'],
-          },
-        ],
-      });
-      debug('fetched living spot request state');
+    let stalledKills = [];
+    let availableAZ = {};
+    let allPricingHistory = {};
+
+    // It would be better to do it by classification than by region for
+    // better concurrency, but this is easier and not too slow considering
+    // the 75s iteration frequency
+    _.forEach(this.ec2, async (obj, region) => {
+      let response = await Promise.all([
```

Comment:
I think this is probably something we should do, but I think it's a little too much for this pull request.